qa/tasks/cephfs/test_mirroring.py

```diff
@@ -14,30 +14,13 @@
 
 log = logging.getLogger(__name__)
 
-class TestMirroring(CephFSTestCase):
-    MDSS_REQUIRED = 5
-    CLIENTS_REQUIRED = 2
-    REQUIRE_BACKUP_FILESYSTEM = True
-
-    MODULE_NAME = "mirroring"
-
-    def setUp(self):
-        super(TestMirroring, self).setUp()
-        self.primary_fs_name = self.fs.name
-        self.primary_fs_id = self.fs.id
-        self.secondary_fs_name = self.backup_fs.name
-        self.secondary_fs_id = self.backup_fs.id
-        self.enable_mirroring_module()
-
-    def tearDown(self):
-        self.disable_mirroring_module()
-        super(TestMirroring, self).tearDown()
 
+class MirroringHelpers(CephFSTestCase):
```

Comment:
it'a a bit complicated, the actual plan is, YES make this helper class an actual helper class but it requires a lot of changes that are unrelated to this PR(making changes to all other test cases). It's a bit hard to explain you in comment, if possible do let know if/when you can join for a call or maybe discuss this post standup tomorrow?

Reply:
Sorry but hard to explain why this needs to be subclassed from `CephFSTestCase`? Could you explain a bit and I'll try my best to understand.

Reply:
Note that it's not just me who is required to review PRs - so taking time to explain here is going to benefit everyone.

Reply:
So the thing is that the change to enable multiple peers required me to create a helper class since the existing helpers in the class `TestMirroring` has hard coded values that only applies to single peer but just adding those helpers into a class and use it into test classes directly is not enough because the helpers that are used in the existing class `TestMirroring` (that are also used in the classes I wrote) now need modifications. This forced me to make changes in this PR that are not related (to the tracker). So in order to make minimal unrelated changes, some helpers still depends on the `CephFSTestCase`, thus you see it being inherited. 

I'm planning to make it a proper helper class but I'm not sure which way to go with; either I can make the changes here in this PR altogether or in a new PR. I'm open to suggestions.

Reply:
> So the thing is that the change to enable multiple peers required me to create a helper class since the existing helpers in the class `TestMirroring` has hard coded values that only applies to single peer but just adding those helpers into a class and use it into test classes directly is not enough because the helpers that are used in the existing class `TestMirroring` (that are also used in the classes I wrote) now need modifications. This forced me to make changes in this PR that are not related (to the tracker). So in order to make minimal unrelated changes, some helpers still depends on the `CephFSTestCase`, thus you see it being inherited.
> 
> I'm planning to make it a proper helper class but I'm not sure which way to go with; either I can make the changes here in this PR altogether or in a new PR. I'm open to suggestions.

IMO having is as a separate (standalone) helper would be the better. I understand that its more work that's the current implementation, but it way better in maintenance in the longer run.

Reply:
so you want me to do it in this PR itself right?

Reply:
yes